framework/python/src/common/risk_profile.py

```diff
@@ -73,6 +73,11 @@ def load(self, profile_json, profile_format):
     self._validate(profile_json, profile_format)
     self.update_risk(profile_format)
 
+    # Check if the profile has expired
+    expired = self._expired()
+    if expired:
+      self.status = 'Expired'
+
```

Comment:
The _validate method already updates the status correctly so if we want to maintain this within the load/constructor process, we should update the _validate method directly to account for this.

```
  def _validate(self, profile_json, profile_format):
    if self._expired():
      self.status = 'Expired'
    elif self._valid(profile_json, profile_format):
      # User only wants to save a draft
      if 'status' in profile_json and profile_json['status'] == 'Draft':
        self.status = 'Draft'
      else:
        self.status = 'Valid'
    else:
      self.status = 'Draft'
```